### PR TITLE
Feature/ex 4827 refactor url module

### PR DESCRIPTION
**Pull request by guillei10**
# Description

The goal of this PR is to refactor some parts of the URL module:

- Change the `updateStoreFromUrl` action into an `ActionClass`
- Change `relatedTags` property to singular to be displayed correctly in the URL.
- Added a `isLoadedFromUrl` flag that controls the flux and avoids undesired loops or duplicated events.

## Motivation

A review and improvement of the URL Module.

## Test

Play with the parameters of the Url (query, related tags, page) and verify everything is set correctly.

### CODE REVIEW

**Comment by tajespasarela** [comment]
When the URL is loaded with query, related tag & page, the page disappears

https://user-images.githubusercontent.com/5759712/135474573-130a943f-73fd-4a9e-8789-6194e02aa7dc.mov

:

File: packages/x-components/src/x-modules/url/store/actions/update-url.action.ts [hunk 1]
```diff
@@ -1,61 +1,107 @@
...
- * @internal
+ * @public
  */
-export const updateUrl: UrlXStoreModule['actions']['updateUrl'] = ({
-  getters: { urlParams, urlMappedParamNames }
-}) => {
-  const url = new URL(window.location.href);
-  const oldParams = url.searchParams;
+export class UpdateUrlAction implements ActionsClass<UrlXStoreModule> {
+  /** Parameters that will require to push the browser state instead of replace. */
```
Comment (pmareke) [inline packages/x-components/src/x-modules/url/store/actions/update-url.action.ts:12]: Why this is a class variable? It's only used inside the `pushableParamsChanged`.

**Review by pmareke — CHANGES_REQUESTED** [review]

File: packages/x-components/src/x-modules/url/events.types.ts [hunk 2]
```diff
@@ -26,15 +26,14 @@ export interface UrlXEvents {
    * Document has finished loading.
    * * Payload: none.
```
Comment (tiborux) [inline packages/x-components/src/x-modules/url/events.types.ts:27]: Update the payload value

File: packages/x-components/src/x-modules/url/store/actions/update-url.action.ts [hunk 3]
```diff
@@ -1,61 +1,107 @@
...
+    });
   }
-};
 
+  /**
+   * Checks if the pushable params have changed.
+   *
+   * @param urlParams - The new Url params.
+   * @param oldValues - The old Url params.
+   * @internal
```
Comment (tiborux) [inline packages/x-components/src/x-modules/url/store/actions/update-url.action.ts:86]: move the internal tag after the return.

File: packages/x-components/src/x-modules/url/wiring.ts [hunk 4]
```diff
@@ -12,6 +13,7 @@ import { createWiring } from '../../wiring/wiring.utils';
  * @internal
  */
 const wireCommit = namespacedWireCommit('url');
+const wireDispatch = namespacedWireDispatch('url');
```
Comment (tiborux) [inline packages/x-components/src/x-modules/url/wiring.ts:16]: Document this

**Review by tiborux — CHANGES_REQUESTED** [review]

File: packages/x-components/src/x-modules/url/store/actions/update-url.action.ts (hunk 1 above)
Comment (guillei10) [reply packages/x-components/src/x-modules/url/store/actions/update-url.action.ts:12]: I followed the same approach we did with the `urlParams` getter. Moving this to a variable to make it more easy to find when we make changes if needed.

File: packages/x-components/src/x-modules/url/components/url-handler.vue [hunk 5]
```diff
@@ -63,17 +57,24 @@
     }
 
     /**
-     * Emits the event with the state values when the document is loaded and open X if there is
-     * a query in the Url.
+     * Checks if the parameter query is in the url.
+     *
+     * @returns Boolean.
```
Comment (pmareke) [inline packages/x-components/src/x-modules/url/components/url-handler.vue:62]: @returns Boolean if the parameter query is in the url.

File: packages/x-components/src/x-modules/url/store/actions/update-url.action.ts (hunk 1 above)
Comment (pmareke) [reply packages/x-components/src/x-modules/url/store/actions/update-url.action.ts:12]: The idea of having a variable is really good, what I meant was to declare inside the method `pushableParamsChanged` as it's the only place where it0s used.

**Review by pmareke — CHANGES_REQUESTED** [review]

File: packages/x-components/src/x-modules/url/store/actions/update-url.action.ts (hunk 1 above)
Comment (guillei10) [reply packages/x-components/src/x-modules/url/store/actions/update-url.action.ts:12]: This way you don't have to look for the variable into a function.

**Review by tiborux — DISMISSED** [review]

**Review by pmareke — APPROVED** [review]

**Review by tiborux — APPROVED** [review]

**Pull request merged**